pkg/hintrunner/hint.go

```diff
@@ -88,3 +90,78 @@ func (hint TestLessThan) Execute(vm *VM.VirtualMachine) error {
 
 	return nil
 }
+
+type WideMul128 struct {
+	lhs  ResOperander
+	rhs  ResOperander
+	high CellRefer
+	low  CellRefer
+}
+
+func (hint WideMul128) String() string {
+	return "WideMul128"
+}
+
+func (hint WideMul128) Execute(vm *VM.VirtualMachine) error {
+	mask := MaxU128Felt()
+
+	lhs, err := hint.lhs.Resolve(vm)
+	if err != nil {
+		return fmt.Errorf("resolve lhs operand %s: %v", hint.lhs, err)
+	}
+
+	rhs, err := hint.rhs.Resolve(vm)
+	if err != nil {
+		return fmt.Errorf("resolve rhs operand %s: %v", hint.rhs, err)
+	}
+
+	lhsFelt, err := lhs.FieldElement()
+	if err != nil {
+		return err
+	}
+	if lhsFelt.Cmp(&mask) > 0 {
+		return fmt.Errorf("lhs operand %s should be u128", lhsFelt)
+	}
+
+	rhsFelt, err := rhs.FieldElement()
+	if err != nil {
+		return err
+	}
+	if rhsFelt.Cmp(&mask) > 0 {
+		return fmt.Errorf("rhs operand %s should be u128", rhsFelt)
+	}
+
+	lhsU256 := uint256.Int(lhsFelt.Bits())
+	rhsU256 := uint256.Int(rhsFelt.Bits())
+	mul := lhsU256.Mul(&lhsU256, &rhsU256)
+
+	bytes := mul.Bytes32()
+
+	low := f.One()
+	low.SetBytes(bytes[16:])
+
+	high := f.One()
+	high.SetBytes(bytes[:16])
+
+	lowCell, err := hint.low.Get(vm)
```

Comment:
Could you please change this vars with suffix `Cell` to suffix `Addr` or similar, since the concept of `Cell` no longer exists and could be confusing